web/blueprints/helpers/table.py

```diff
@@ -258,3 +342,30 @@ def datetime_format(dt, default=None):
             'formatted': default if default is not None else datetime_filter(None),
             'timestamp': None,
         }
+
+
+def enforce_url_params(url, params):
+    """Safely enforce query values in an url
+
+    :param str url: The url to patch
+    :param dict params: The parameters to enforce in the URL query
+        part
+    """
+    if not params:
+        return url
+    # we need to use a list because of mutability
+    url_parts = list(urlparse(url))
+    query_parts = dict(parse_qsl(url_parts[4]))
+    query_parts.update(params)
+    url_parts[4] = urlencode(query_parts)
+    return urlunparse(url_parts)
+
+
+@lazy_join
+def button_toolbar(title: str, href: str, icon: str = "glyphicon-plus")\
+        -> LazilyJoined:
+    params = html_params(class_="btn btn-primary", href=href)
+    yield f"<a {params}>"
+    yield f"<span class=\"glyphicon {icon}\"></span>"
```

Comment:
There should be a space after the icon.

Reply:
What do you mean „after the icon“? After the `</span>`, or inbetween, or after `{icon}`?

Reply:
Resolved in f7b046d8a4c51784ea5618c31427cbcd0d14d0c5